tidy3d/components/simulation.py

```diff
@@ -1258,7 +1258,8 @@ def get_dls(geom: Geometry, axis: Axis, num_dls: int) -> List[float]:
         def snap_to_grid(geom: Geometry, axis: Axis) -> Geometry:
             """Snap a 2D material to the Yee grid."""
             center = get_bounds(geom, axis)[0]
-            assert get_bounds(geom, axis)[0] == get_bounds(geom, axis)[1]
+            if get_bounds(geom, axis)[0] != get_bounds(geom, axis)[1]:
+                raise AssertionError
```

Comment:
@caseyflex Should we print a nice message here?

Reply:
"Unexpected error encountered while processing 2D material. The upper and lower bounds of the geometry in the normal direction are not equal. If you encounter this error, please create an issue in the Tidy3D github repository."

Shashwat added this assertion. I do not think it would ever arise in practice

Reply:
Great, thanks a lot!